tests/testsuite/lto.rs

```diff
@@ -622,7 +622,7 @@ fn dylib() {
 [FRESH] registry-shared v0.0.1
 [COMPILING] registry v0.0.1
 [RUNNING] `rustc --crate-name registry [..]
-[DIRTY] bar v0.0.0 ([..]): dependency info changed
+[DIRTY] bar v0.0.0 ([..]): dependency info changed (registry_shared)
```

Comment:
This message looks strange to me. Should it not output `registry` rather than `registry_shared` here?

Reply:
Part of the problem is we only report the first dependency, rather than all.

I suspect what is happening here is we are reusing `registry_shared` with a version from a couple builds back.  This is a different hash than the immediately prior build, so it requires rebuilding `bar` but this isn't quite clear without more context.

I wonder if this change will cause more confusion than good and we should defer to #2904 which would add a `cargo report` subcommand for reporting why the last command rebuilt and we can include a lot more context.

Reply:
See <https://github.com/rust-lang/cargo/issues/13859#issuecomment-2103227615>, yet it might need some design discussions before proceed.